lively.ide/studio/flap.cp.js

```diff
@@ -0,0 +1,77 @@
+import { Color, pt } from 'lively.graphics';
+import { TilingLayout, component, ViewModel } from 'lively.morphic';
+import { rect } from 'lively.graphics/geometry-2d.js';
+
+export class FlapModel extends ViewModel {
+  static get properties () {
+    return {
+      target: {
+        type: 'String'
+      },
+      action: { },
+      relayoutRoutine: {},
+      openingRoutine: {},
+      bindings: {
+        get () {
+          return [
+            { signal: 'onMouseDown', handler: 'executeAction' },
+            { signal: 'openInWorld', handler: 'openInWorld' }
+          ];
+        }
+      },
+      expose: {
+        get () {
+          return ['onWorldResize', 'isFlap'];
+        }
+      }
+    };
+  }
+
+  get isFlap () {
+    return true;
+  }
+
+  async executeAction () {
+    this.action.bind(this)();
+  }
+
+  onWorldResize () {
+    this.relayoutRoutine.bind(this)();
+  }
+
+  openInWorld () {
+    this.openingRoutine.bind(this)();
```

Comment:
Can we just use these as plain callbacks without binding them dynamically agains the view model? We can just pass the view model as an argument to the callbacks. That way the callbacks being defined outside of the view model class could contain a parameter name like `flapViewModel` which makes things more clear than implicitly assuming this refers to the flap view model.